lib/spack/spack/test/util/compression.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import re
+import sys
+from typing import Dict  # novm
+
+import pytest
+
+from llnl.util.filesystem import working_dir
+
+from spack.util import compression as scomp
+from spack.util.executable import CommandNotFoundError, which
+
+
+def compose_to_dict(pkg, ext, dict):
+    dict[ext] = pkg
+    return pkg
+
+
+ext_archive = {}  # type: Dict[str, str]
+fake_archives = [compose_to_dict('.'.join(['Foo', ext]), ext, ext_archive) for ext
+                 in scomp.ALLOWED_ARCHIVE_TYPES]
+
+
+def build_temp_archive(extension, mode, archiver, compression, archive_file, ext_dir):
+    if compression.lower() == ':z':
```

Comment:
It looks like `archive_file` is prepending a `:` to get compression - why is that? Ideally when it comes to constructing and checking options IMO enums are more readable than string constructions (minor typos can lead to hard-to-debug errors when options are selected with strings).